packages/electron/README.md

```diff
@@ -0,0 +1,504 @@
+
+### Attributes
+
+Custom attributes are key-value pairs that can be added to your error reports. They are used in report aggregation,
+sorting and filtering, can provide better contextual data for an error, and much more. They are foundational to many of
+the advanced Backtrace features detailed in
+[Error Reporting documentation](https://docs.saucelabs.com/error-reporting/getting-started/). By default attributes such
+as application name and version are populated automatically based on your package.json information. If Backtrace cannot
+find them, you need to provide them manually via userAttributes attributes.
+
+There are several places where attributes can be added, modified or deleted.
+
+#### Attach attributes object to BacktraceClient
+
+It is possible to include an attributes object during [BacktraceClient](#backtraceclient) initialization. This list of
+attributes will be included with every error report, referred to as global attributes.
+
+```ts
+// Create an attributes object that can be modified throughout runtime
+const attributes: Record<string, unknown> = {
+    release: 'PROD',
+};
+
+// BacktraceClientOptions
+const options: BacktraceConfiguration = {
+    url: 'https://submit.backtrace.io/<universe>/<token>/json',
+
+    // Attach the attributes object
+    userAttributes: attributes,
+};
+
+// Initialize the client
+const client = BacktraceClient.initialize(options);
+```
+
+You can also include attributes that will be resolved when creating a report:
+
+```ts
+// BacktraceClientOptions
+const options: BacktraceConfiguration = {
+    url: 'https://submit.backtrace.io/<universe>/<token>/json',
+
+    // Attach the attributes object
+    userAttributes: () => ({
+        attribute: getAttributeValue(),
+    }),
+};
+
+// Initialize the client
+const client = BacktraceClient.initialize(options);
+```
+
+#### Add attributes during application runtime
+
+Global attributes can be set during the runtime once specific data has be loaded (e.g. a user has logged in).
+
+```ts
+const client = BacktraceClient.initialize(options);
+...
+
+client.addAttribute({
+    "clientID": "de6faf4d-d5b5-486c-9789-318f58a14476"
+})
+```
+
+You can also add attributes that will be resolved when creating a report:
+
+```ts
+const client = BacktraceClient.initialize(options);
+...
+
+client.addAttribute(() => ({
+    "clientID": resolveCurrentClientId()
+}))
+```
+
+#### Add attributes to an error report
+
+The attributes list of a BacktraceReport object can be directly modified.
+
+```ts
+const report: BacktraceReport = new BacktraceReport('My error message', { myReportKey: 'myValue' });
+report.attributes['myReportKey'] = 'New value';
+```
+
+---
+
+### File Attachments
+
+Files can be attached to error reports. This can be done when initalizing the BacktraceClient, updating the
+BacktraceClient, or dynamically for specific reports. When including attachments in BacktraceClient, all files will be
+uploaded with each report.
```

Comment:
I think its worth to mention we support file attachments in the renderer process - or actually do we? How we can pass file attachments from renderer?

Reply:
The file attachments are passed, but only when the error is thrown in the renderer. Attaching renderer attachments when an error is thrown in main is a bit more complicated, and it is not implemented here.

Reply:
Can we note this here? Also if someone passes in memory attachment to the send method, does it mean we won't pass it to main?

Reply:
Memory attachment? Generally, I try to send them anyway by serializing them to JSON.

![image](https://github.com/backtrace-labs/backtrace-javascript/assets/9085864/d1866155-cba3-4a19-8eb4-9a51506d058b)

Added a section in "Integrate the SDK in the renderer process" about the attachments.